bob/src/cluster/quorum.rs

```diff
@@ -124,6 +157,39 @@ impl Quorum {
         put_at_least(key, data, target_nodes, at_least, PutOptions::new_local()).await
     }
 
+    pub(crate) async fn delete_at_remote_nodes(
+        &self,
+        key: BobKey,
+        with_aliens: bool,
+    ) -> Vec<NodeOutput<Error>> {
+        let local_node = self.mapper.local_node_name();
+        let mut target_nodes = vec![];
+        if with_aliens {
+            target_nodes.extend(self.mapper.nodes().values());
+        } else {
+            target_nodes.extend(self.mapper.get_target_nodes_for_key(key))
+        };
+        debug!(
+            "DELETE[{}] cluster quorum put remote nodes {} total target nodes",
+            key,
+            target_nodes.len(),
+        );
+        let count = target_nodes
+            .iter()
+            .filter(|n| n.name() != local_node)
+            .count();
+        let target_nodes = target_nodes
+            .into_iter()
+            .filter(|node| node.name() != local_node);
```

Comment:
Probably it is better to remove one element from the vector than to repeat the filtering 2 times